src/style/utils/merge-deep.spec.js

```diff
@@ -0,0 +1,44 @@
+import { mergeDeep } from './merge-deep';
+
+describe('mergeDeep', () => {
+  let base;
+
+  beforeEach(() => {
+    base = {
+      a: 1, b: { nestedB: 1 }
+    };
+  });
+
+  const toMerge = {
+    a: 1, b: { nestedA: 1 }
+  };
+
+  const objectToMergeWithSameKey = {
+    a: 2
+  };
+
+  const objectWithArray = {
+    c: ['inArray']
+  };
+
+  it('merges objects with nested keys', () => {
+    expect(mergeDeep(base, toMerge)).toEqual({ a: 1, b: { nestedA: 1, nestedB: 1 } });
+  });
+
+  it('merges the objects and creates a new key', () => {
+    const newKeyObj = { a: 1, b: { nestedB: 1 }, c: 'test' };
+    expect(mergeDeep(base, newKeyObj)).toEqual({ a: 1, b: { nestedB: 1 }, c: 'test' });
+  });
+
+  it('merges objects and overwrites keys if they exist', () => {
+    expect(mergeDeep(base, objectToMergeWithSameKey)).toEqual({ a: 2, b: { nestedB: 1 } });
+  });
+
+  it('handles the merging of objects which contain arrays', () => {
+    expect(mergeDeep(base, objectWithArray)).toEqual({ a: 1, b: { nestedB: 1 }, c: ['inArray'] });
+  });
```

Comment:
I think that we need a test that checks if arrays are merged properly - e.g.: mergeDeep({ a: [1,2] }, { a: [3] }) should result in { a: [1,2,3] }

Reply:
Do we want to allow this behaviour? All the other values are overwritten when they have the same key. i.e. if `mergeDeep({ a: 1 }, { a: 2 }) => { a: 2 }`, should it not follow that `mergeDeep({ a: [1] }, { a: [2] }) => { a: [2] }`.

Reply:
You are right, I'll leave that as resolved.
I was using https://github.com/TehShrike/deepmerge, which does that by default and that behaviour is configurable , but on the other side lodash's ```_.merge({ a: [1,2] }, { a: [3] })``` will result in ```{ a: [3, 2] }```.